### PR TITLE
FIX Implement support new league/flysystem v3

### PR DESCRIPTION
**Pull request by sabina-talipova**
### Description
- Plugins classes were removed from `league/flysystem` version 2 without any replacement([see docs](https://flysystem.thephpleague.com/docs/upgrade-from-1.x/)). 
- `getImportFiles ` method was updated to support new  changes in `league/flysystem` [see docs](https://flysystem.thephpleague.com/docs/upgrade-from-1.x/)

### Parent issue
- https://github.com/silverstripeltd/product-issues/issues/676

### Fix issue
- https://github.com/silverstripe/silverstripe-registry/actions/runs/4130270901/jobs/7159269470

### CODE REVIEW

File: src/RegistryImportFeed.php [hunk 1]
```diff
@@ -80,7 +78,7 @@ public function getLatest()
      */
     public function setAssetHandler(GeneratedAssetHandler $handler)
     {
-        $handler->getFilesystem()->addPlugin(new ListFiles);
+        $handler->getFilesystem();
```
Comment (ChloeSartorelli) [inline src/RegistryImportFeed.php]:
```suggestion
```
This doesn't actually do anything, just remove it entirely.

File: src/RegistryImportFeed.php [hunk 2]
```diff
@@ -117,16 +115,16 @@ public function getStoragePath($modelClass = null)
...
 
         foreach ($importFiles as $importFile) {
             $files->push(RegistryImportFeedEntry::create(
-                $importFile['basename'],
+                basename($importFile->path()),
                 '',
-                DBDatetime::create()->setValue($importFile['timestamp'])->Format(DBDatetime::ISO_DATETIME),
-                $this->getAssetsDir() . '/' . $importFile['path']
+                DBDatetime::create()->setValue($importFile->lastModified())->Format(DBDatetime::ISO_DATETIME),
+                $this->getAssetsDir() . '/' . $importFile->path()
```
Comment (ChloeSartorelli) [inline src/RegistryImportFeed.php:125]: What are all of these changes doing? Why are these necessary? They don't seem relevant to removing a call to `addPlugin()`?
Or is this replacing the functionality that plugin was applying?

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
There are more changes happening here than the commit message implies. Please use a more accurate commit message.

File: src/RegistryImportFeed.php (hunk 2 above)
Comment (sabina-talipova) [reply src/RegistryImportFeed.php:125]: All this changes have been done to support new version of `league/flysystem`. I've added test especially for this method.

File: src/RegistryImportFeed.php (hunk 1 above)
Comment (sabina-talipova) [reply src/RegistryImportFeed.php]: DONE

**Comment by GuySartorelli** [comment]
@sabina-talipova please update commit message as per https://github.com/silverstripe/silverstripe-registry/pull/77#pullrequestreview-1298674591

You can change the commit message with `git commit --amend` and then force push

**Review by GuySartorelli — APPROVED** [review]

**Comment by GuySartorelli** [comment]
Merge when commit message is updated.
